src/bgw/job.c

```diff
@@ -1204,6 +1262,12 @@ ts_bgw_job_entrypoint(PG_FUNCTION_ARGS)
 	 */
 	ts_bgw_job_stat_mark_end(job, res, NULL);
 
+	if (!job_failed && ts_is_tss_enabled() && scheduler_test_hook == NULL)
+	{
+		const char *stmt = ts_bgw_job_function_call_string(job);
```

Comment:
This function will fail for some jobs (e.g., the telemetry job and the error log retention job) so it will cause `ts_bgw_job_entrypoint` to fail with an error, which will cause the scheduler to re-start the job because of the failure.

I would argue that you need to make sure that it behaves sanely even for cases where there are no processes associated with the job and not throw an error, since this will cause further problems for the job scheduler and make for confusing error reporting.

Reply:
I added a check to fallback to PROKIND_FUNC in case of don't find the function/procedure so it will not raise an exception